base/broadcast.jl

```diff
@@ -137,7 +137,7 @@ BroadcastStyle(a::AbstractArrayStyle, ::Style{Tuple})    = a
 BroadcastStyle(::A, ::A) where A<:ArrayStyle             = A()
 BroadcastStyle(::ArrayStyle, ::ArrayStyle)               = Unknown()
 BroadcastStyle(::A, ::A) where A<:AbstractArrayStyle     = A()
-Base.@pure function BroadcastStyle(a::A, b::B) where {A<:AbstractArrayStyle{M},B<:AbstractArrayStyle{N}} where {M,N}
+Base.@assume_effects :total function BroadcastStyle(a::A, b::B) where {A<:AbstractArrayStyle{M},B<:AbstractArrayStyle{N}} where {M,N}
```

Comment:
Do we even need this annotation anymore? It assumes that `max` is total, which is normally true, but not guaranteed

Reply:
I confirmed that we don't need this setting annotation to prove the concrete-eval eligibility of this method. Maybe we want to merge #44785 first so that we can test this eligibility in our test suite.

Reply:
The annotation is removed, and now we explicitly test the totality of this method.